FABulous/fabric_generator/fabric_gen.py

```diff
@@ -1990,69 +2000,116 @@ def generateFabric(self) -> None:
                             pre = f"Tile_X{i}Y{j}_"
                         else:
                             pre = ""
-                        if tile.globalConfigBits > 0 or superTile:
-                            # frameData signal
-                            if x == 0:
-                                portsPairs.append(
-                                    (f"{pre}FrameData", f"Tile_Y{y}_FrameData")
-                                )
 
-                            elif (x + i - 1, y + j) not in superTileLoc:
-                                portsPairs.append(
-                                    (
-                                        f"{pre}FrameData",
-                                        f"Tile_X{x+i-1}Y{y+j}_FrameData_O",
-                                    )
-                                )
+                        supertile_x = x + i
+                        supertile_y = y + j
 
-                            # frameData_O signal
-                            if x == len(self.fabric.tile[0]) - 1:
-                                portsPairs.append(
-                                    (f"{pre}FrameData_O", f"Tile_X{x}Y{y}_FrameData_O")
-                                )
+                        # Connect the FrameData port to the previous tiles'
+                        # FrameData_O signals.
+                        # If the previous tile is NULL, continue the search.
+                        # If all previous tiles are NULL, connect to the fabrics
+                        # Tile_Y{y}_FrameData signals.
 
-                            elif (x + i - 1, y + j) not in superTileLoc:
-                                portsPairs.append(
-                                    (
-                                        f"{pre}FrameData_O",
-                                        f"Tile_X{x+i}Y{y+j}_FrameData_O",
-                                    )
+                        done = False
+
+                        # Get all x-positions before this tile
```

Comment:
```suggestion
                        # Get all x-positions to the west of this tile
```

Maybe this a bit more precise?

Reply:
Yes, good point! I also updated the other comments.